### PR TITLE
Feat/proposal budget

### PR DESCRIPTION
**Pull request by 1emu**
## Contested budget endpoint for a proposal: 

Returns the budget for the received proposal, along with the contested budget for all the active proposals within the budget start and end date

- If there is only one active proposal for that category and budget

image

- If there is another competing proposal:

image
(*) Notice the contested percentage can be higher than 100%

- The rest of the budget info looks like this:
image

## Proposal Budget section

- With requested budget over total category budget and total category budget cards.

image
image
image

-  Created a `DistributionBar` component, that receives a list of items, each with its value, style, and optional popup properties.

## Grant Categories icons

image

image

### CODE REVIEW

File: src/components/Common/DistributionBar/DistributionBarItem.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  style: string
+  selected?: boolean
+  popupContent?: DistributionBarPopupContent
+}
+
+const DistributionBarItem = ({ value, style, selected, popupContent, total }: Props) => {
+  const valuePercentage = getFormattedPercentage(value, total)
+  const distributionBar = (
+    <div
+      className={TokenList.join(['DistributionBarItem', !!selected && 'DistributionBarItem__Selected', style])}
```
Comment (andyesp) [inline src/components/Common/DistributionBar/DistributionBarItem.tsx:22]: If it's a modifier then it's `--selected`.

File: src/components/Common/DistributionBar/DistributionBarItem.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  value: number
+  total: number
+  style: string
+  selected?: boolean
+  popupContent?: DistributionBarPopupContent
+}
+
+const DistributionBarItem = ({ value, style, selected, popupContent, total }: Props) => {
+  const valuePercentage = getFormattedPercentage(value, total)
+  const distributionBar = (
```
Comment (andyesp) [inline src/components/Common/DistributionBar/DistributionBarItem.tsx:20]: Can you move this to a `DistributionBar` component?

File: src/components/Grants/Current/BudgetBanner.css [hunk 3]
```diff
@@ -45,3 +49,14 @@
...
 }
+
+.BudgetBanner__LabelWithIcon {
+  display: flex;
+  gap: 20px;
+  align-items: center;
+  height: 50px;
+}
+
+.BudgetBannerItem {
```
Comment (andyesp) [inline src/components/Grants/Current/BudgetBanner.css:60]: This belongs in `BudgetBannerItem.css`

File: src/components/Proposal/View/Budget/RequestedBudgetCard.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+export default function RequestedBudgetCard({ proposal, budget }: Props) {
+  const t = useFormatMessage()
+  const grantCategory = proposal.configuration.category
+  const totalCategoryBudget = budget.categories[snakeCase(grantCategory)]?.total || 0
+  const requestedBudget = proposal.configuration.size
+  const remainingTotalBudget = totalCategoryBudget - requestedBudget
+  const remainingTotalBudgetDisplayed = remainingTotalBudget > 0 ? remainingTotalBudget : 0
+
+  const items: DistributionItemProps[] = [
```
Comment (andyesp) [inline src/components/Proposal/View/Budget/RequestedBudgetCard.tsx:27]: Add a `useMemo` here

File: src/components/Proposal/View/Budget/RequestedBudgetCard.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    {
+      value: requestedBudget,
+      style: 'RequestedBudgetBar',
```
Comment (andyesp) [inline src/components/Proposal/View/Budget/RequestedBudgetCard.tsx:30]: If this is the className I'd rename `style` to `class` or `className`.

File: src/components/Proposal/View/Budget/RequestedBudgetCard.css [hunk 6]
```diff
@@ -0,0 +1,20 @@
+.RequestedBudgetCard {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  gap: 8px;
+  margin-bottom: 2px;
+}
+
+.RequestedBudgetBar {
```
Comment (andyesp) [inline src/components/Proposal/View/Budget/RequestedBudgetCard.css:9]: I'd move this to the DistributionBar css file

File: src/components/Proposal/View/Budget/RequestedBudgetCard.css [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  background: linear-gradient(90deg, #017f0a 0%, #00bfd9 100.66%);
+  border-radius: 5px;
+  box-shadow: 0 1px 4px rgba(0, 182, 73, 0.4);
+  height: 8px;
+}
+
+.TotalBudgetBar {
```
Comment (andyesp) [inline src/components/Proposal/View/Budget/RequestedBudgetCard.css:16]: I'd move this to the DistributionBar file css too

**Review by andyesp — CHANGES_REQUESTED** [review]

File: src/components/Proposal/View/Budget/RequestedBudgetCard.css (hunk 6 above)
Comment (1emu) [reply src/components/Proposal/View/Budget/RequestedBudgetCard.css:9]: This are classes particular to how the distribution bar looks on this `RequestedBudgetCard`. 
If some other component that uses the distribution bar, it should declare different styles (there are no defaults yet)

**Comment by 1emu** [comment]
### New changes:

## Show Allocated budget

image

image


## Show when proposal is Over Budget 

image

**Review by andyesp — APPROVED** [review]

**Pull request merged**